client/src/components/profile/AuthenticationCard.vue

```diff
@@ -62,6 +62,7 @@ const props = defineProps<{
   state: AuthenticationCardState;
   authMethod: DeviceSaveStrategyTag | CustomDeviceSaveStrategyTag;
   disabled?: boolean;
+  hideCheckmark?: boolean;
```

Comment:
I don't like props like this. The parent shouldn't have to worry about the child displaying a checkbox or not, it makes the parent too aware of its children structure.

The prop should be a some kind of general thing and the component determines that if it's in this state, it should display or not a checkmark. For example, on Github for each commit you have a verified checkmark: the prop should be something like `isVerified: boolean`, not `displayCheckmark`.